packages/components/src/FormGroup/FormGroup.tsx

```diff
@@ -0,0 +1,107 @@
+import clsx from "clsx";
+import React, { ElementType, ReactNode } from "react";
+import Text from "../Text";
+import styles from "./FormGroup.module.css";
+
+type FormGroupProps = {
+  /**
+   * The contents of the form group. We suggest a FormGroup.Title followed by
+   * interactive elements.
+   *
+   * Should be wrapped in a fragment to allow our styling to control the spacing
+   * between elements.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'fieldset'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+type TitleProps = {
+  /**
+   * The title of the form group.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the wrapper should be.
+   * @default 'legend'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+type ItemsProps = {
+  /**
+   * The content of the helper form elements in the form group.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'div'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+/**
+ * Helper sub-component for styling the title at the top of the component.
+ */
+const Title = ({
+  children,
+  as: Component = "legend",
+  className,
+}: TitleProps) => {
+  return (
+    <Component className={clsx(className, styles.title)}>
+      <Text size="sm" weight="bold">
+        {children}
+      </Text>
+    </Component>
+  );
+};
+
+/**
+ * Helper sub-component for styling the form elements in the component.
+ */
+const Items = ({ children, as: Component = "div", className }: ItemsProps) => {
```

Comment:
I'm surprised to see the `Component` annotation here, since `ItemProps` defines `as` as `ElementType`.

Do we need that because TypeScript isn't happy with trying to do `<Component>` with intrinsic JSX elements or something?

Reply:
In this instance I believe `as: Component` is just renaming the variable and not providing a type annotation. Then you can use `<Component .../>` instead of `<as ... />`

Reply:
Yep, you're right. It's not even possible to have a type annotation in the middle of object destructuring like that.